daemon/daemon.go

```diff
@@ -963,8 +963,11 @@ func (daemon *Daemon) initDiscovery(config *Config) error {
 // - Daemon live restore
 func (daemon *Daemon) Reload(config *Config) error {
 	var err error
-	// used to hold reloaded changes
-	attributes := map[string]string{}
+
+	daemon.configStore.reloadLock.Lock()
+	defer daemon.configStore.reloadLock.Unlock()
```

Comment:
Reordering the locking means that `reloadLock` will be held through the call to `LogDaemonEventWithAttributes`, which can block for up to 100 ms. I'm not sure it matters, but it seems worth pointing out.